service/Abstractions/AI/ITextGenerator.cs

```diff
@@ -19,7 +20,7 @@ public interface ITextGenerator : ITextTokenizer
     /// <param name="options">Options for the LLM request</param>
     /// <param name="cancellationToken">Async task cancellation token</param>
     /// <returns>Text generated, returned as a stream of strings/tokens</returns>
-    public IAsyncEnumerable<string> GenerateTextAsync(
+    public IAsyncEnumerable<(string? Text, TokenUsage? TokenUsage)> GenerateTextAsync(
```

Comment:
I believe the convention is camelCase for the names. In any case, I think it would be more future proof streaming a single object instead of multiple params, something like:

`public IAsyncEnumerable<TextContent> GenerateTextAsync(...)`
 
where `TextContent` is a new class that contains text, token usage and future stuff if needed.